Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -460,6 +527,68 @@ void HTMLAttachmentElement::setFile(RefPtr<File>&& file, UpdateDisplayAttributes
     invalidateRendering();
 }
 
+#if ATTACHMENT_LOG_DOCUMENT_TRAFFIC
+class AttachmentEvent {
+public:
+    const void* attachment() const { return m_attachment; }
+    const void* document() const { return m_document; }
+    String uniqueIdentifier() const { return m_uniqueIdentifier; }
+    WTF::MonotonicTime time() const { return m_time; }
+    StackTrace& stackTrace() const { return *m_stackTrace; }
+
+    void capture(const HTMLAttachmentElement& a, WTF::MonotonicTime t)
+    {
+        m_attachment = &a;
+        m_document = &a.document();
+        m_uniqueIdentifier = a.uniqueIdentifier();
+        ASSERT(!!t);
+        m_time = t;
+        m_stackTrace = StackTrace::captureStackTrace(64);
+    }
+
+    void reset()
+    {
+        m_attachment = nullptr;
+        m_stackTrace = nullptr;
+    }
+
+    explicit operator bool() const
+    {
+        ASSERT(!m_attachment == !m_stackTrace);
+        return !!m_attachment;
+    }
+
+private:
+    const void* m_attachment { nullptr };
+    const void* m_document { nullptr };
```

Comment:
I don't think using raw pointers is okay.

Reply:
They're only used in comparisons, they're never dereferenced.
But I'll use `uintptr_t` instead, to reduce the potential for misuse.